parse.js

```diff
@@ -3,19 +3,26 @@
 const util = require('util');
 const fs = require('fs');
 const simpleParser = require('mailparser').simpleParser;
-var mailattach = {
-    filename: [],
-    content: []
-}
+var mailattach = []
 
 // mailattch = [{ filename: '', content: '' }, {  }]
+// regex to get email address inside the brackets <>. 
+// Because mail.email returns email: 'Hello world! <helloworld@gmail.com>'
+
+var regExp = /\<([^)]+)\>/;
 
 module.exports = function mailparser(data){
-    console.log("here");
     return new Promise(function(resolve, reject){
         simpleParser(data)
             .then(mail => {
-            resolve(mail.attachments);
+                for (var attach in mail.attachments){
+                mailattach.push({
+                    attachment : mail.attachments[attach],
+                    date : mail.date,
+                    email : regExp.exec(mail.from.text)[1]
```

Comment:
is it guaranteed that the email always are always enclosed with angle brackets?

Reply:
I have checked for gmail and yahoo mail. it stores it in the <> brackets. I will look into this !

Reply:
I have edited the regex to accept both angle and square brackets